app/state/recurringReservations.js

```diff
@@ -89,6 +92,13 @@ const recurringReservationsReducer = handleActions({
   [actions.changeNumberOfOccurrences]: (state, action) => adjustState({
     ...state, numberOfOccurrences: parseInt(action.payload, 10),
   }),
+  [actionTypes.UI.CLOSE_MODAL]: (state, action) => {
+    const modalType = action.payload;
+    if (modalType === modalTypes.RESERVATION_SUCCESS) {
+      return initialState;
+    }
+    return state;
+  },
```

Comment:
Good point.

- [ ] I think this should happen when RESERVATION_CONFIRM modal type gets closed as well.